app/controllers/chat_channel_memberships_controller.rb

```diff
@@ -39,6 +39,24 @@ def create
     redirect_to edit_chat_channel_membership_path(membership)
   end
 
+  def join_channel
+    membership_params = params[:chat_channel_membership]
+    chat_channel = ChatChannel.find(membership_params[:chat_channel_id])
+    user = User.find_by(id: membership_params[:user_id])
+    existing_membership = ChatChannelMembership.find_by(user_id: user.id, chat_channel_id: chat_channel.id)
+    if existing_membership.present? && %w[active joining_request].exclude?(existing_membership.status)
+      success = existing_membership.update(status: "joining_request", role: "member") ? true : false
```

Comment:
`update` will return true or false so there is no need for the extra ternary statement.

Also, if the membership already exists then aren't we done? Why would we mark a current membership as a "joining_request"?

Reply:
There might be few cases like if a person has left the channel or removed from the channel so in that case, if they again want to join the closed group they can send the request.
That's why we're looking for existing memberships and updating its status  as "joining_request"